tests/test_models/test_holt_winters_model.py

```diff
@@ -119,3 +123,155 @@ def test_get_model_after_training(example_tsds, etna_model_class, expected_class
+        model._check_mul_components()
+
+
+def test_predict_components_not_fitted_error(seasonal_dfs):
+    _, df = seasonal_dfs
+    model = _HoltWintersAdapter()
+    with pytest.raises(ValueError, match="This model is not fitted!"):
+        model.predict_components(df=df)
+
+
+def test_forecast_components_not_fitted_error(seasonal_dfs):
+    _, df = seasonal_dfs
+    model = _HoltWintersAdapter()
+    with pytest.raises(ValueError, match="This model is not fitted!"):
+        model.forecast_components(df=df)
+
+
+@pytest.mark.parametrize("trend,trend_error", (("mul", True), ("add", False), (None, False)))
+@pytest.mark.parametrize("seasonal,seasonal_error", (("mul", True), ("add", False), (None, False)))
+def test_check_mul_components(seasonal_dfs, trend, trend_error, seasonal, seasonal_error):
+    _, df = seasonal_dfs
+    model = _HoltWintersAdapter(trend=trend, seasonal=seasonal)
+    model.fit(df, [])
+
+    if trend_error or seasonal_error:
+        context = pytest.raises(ValueError, match="Forecast decomposition is only supported for additive components!")
+    else:
+        context = nullcontext()
+
+    with context:
+        model._check_mul_components()
+
+
+@pytest.mark.parametrize("trend,trend_component", (("add", ["target_component_trend"]), (None, [])))
+@pytest.mark.parametrize("seasonal,seasonal_component", (("add", ["target_component_seasonality"]), (None, [])))
+def test_predict_components_names(seasonal_dfs, trend, trend_component, seasonal, seasonal_component):
+    components_names = set(trend_component + seasonal_component + ["target_component_level"])
+    _, df = seasonal_dfs
+    model = _HoltWintersAdapter(trend=trend, seasonal=seasonal)
+    model.fit(df, [])
+    components = model.predict_components(df)
+    assert set(components.columns) == components_names
+
+
+@pytest.mark.parametrize("trend,trend_component", (("add", ["target_component_trend"]), (None, [])))
+@pytest.mark.parametrize("seasonal,seasonal_component", (("add", ["target_component_seasonality"]), (None, [])))
+def test_forecast_components_names(seasonal_dfs, trend, trend_component, seasonal, seasonal_component):
+    components_names = set(trend_component + seasonal_component + ["target_component_level"])
+    _, df = seasonal_dfs
+    model = _HoltWintersAdapter(trend=trend, seasonal=seasonal)
+    model.fit(df, [])
+    components = model.forecast_components(df)
+    assert set(components.columns) == components_names
+
+
+@pytest.mark.parametrize("df_names", ("seasonal_dfs", "multi_trend_dfs"))
+@pytest.mark.parametrize("trend", ("add", None))
+@pytest.mark.parametrize("seasonal", ("add", None))
+@pytest.mark.parametrize("damped_trend", (True, False))
+@pytest.mark.parametrize("use_boxcox", (True, False))
+def test_predict_components_sum_up_to_target(df_names, trend, seasonal, damped_trend, use_boxcox, request):
+    dfs = request.getfixturevalue(df_names)
+    train, test = dfs
+
+    if trend is None:
```

Comment:
Let't put this logic inside parametrization, just add all the valid cases in line with "trend" parameter